src/main/java/org/folio/fqm/repository/IdStreamer.java

```diff
@@ -90,23 +88,29 @@ public List<UUID> getSortedIds(String derivedTableName,
     }
 
   private int streamIdsInBatch(EntityType entityType,
-                               String derivedTableName,
                                boolean sortResults,
                                Condition sqlWhereClause,
                                int batchSize,
                                Consumer<IdsWithCancelCallback> idsConsumer) {
     return jooqContext.transactionResult(ctx -> {
+      String idValueGetter = entityType
+        .getColumns()
+        .stream()
+        .filter(col -> col.getName().equals(ID_FIELD_NAME))
```

Comment:
Tiniest of nitpicks:
```suggestion
        .filter(col -> ID_FIELD_NAME.equals(col.getName()))
```
That way, we're protected from a potential `NullPointerException` if `col.getName()` is `null` somehow. It also makes it so every call to `.equals()` happens in `ID_FIELD_NAME` which slightly increases the odds that the JVM can optimize it at runtime somehow